OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseTabsDashboardFragment.kt

```diff
@@ -0,0 +1,886 @@
+     * @return `true` if the specified tab fragment needs to be selected, `false` otherwise
+     */
+    private fun shouldSelectFragment(
+        item: FragmentItemModel,
+        @ScreenDef screenName: String
+    ): Boolean {
+        return (screenName == Screen.COURSE_VIDEOS && item.title == getString(R.string.videos_title)) ||
+                (screenName == Screen.COURSE_DISCUSSION && item.title == getString(R.string.discussion_title)) ||
+                (screenName == Screen.DISCUSSION_POST && item.title == getString(R.string.discussion_title)) ||
+                (screenName == Screen.DISCUSSION_TOPIC && item.title == getString(R.string.discussion_title)) ||
+                (screenName == Screen.COURSE_DATES && item.title == getString(R.string.label_dates)) ||
+                (screenName == Screen.COURSE_HANDOUT && item.title == getString(R.string.handouts_title)) ||
+                (screenName == Screen.COURSE_ANNOUNCEMENT && item.title == getString(R.string.announcement_title))
+    }
+
+    private fun setViewPager() {
+        binding.pager.visibility = View.VISIBLE
+        enforceSingleScrollDirection(binding.pager)
+
+        binding.toolbar.tabs.addOnTabSelectedListener(object : OnTabSelectedListener {
+            override fun onTabSelected(tab: TabLayout.Tab) {
+                binding.pager.currentItem = tab.position
+            }
+
+            override fun onTabUnselected(tab: TabLayout.Tab) {}
+            override fun onTabReselected(tab: TabLayout.Tab) {}
+        })
+        binding.pager.adapter = FragmentItemPagerAdapter(requireActivity(), courseTabItems)
+        binding.pager.registerOnPageChangeCallback(object : OnPageChangeCallback() {
+            override fun onPageSelected(position: Int) {
+                super.onPageSelected(position)
+                val item = courseTabItems[position]
+                if (item.listener != null) {
+                    item.listener?.onFragmentSelected()
+                }
+            }
+        })
+
+        TabLayoutMediator(
+            binding.toolbar.tabs,
+            binding.pager
+        ) { tab: TabLayout.Tab, position: Int ->
+            tab.text = courseTabItems[position].title
+        }.attach()
+
+        if (courseTabItems.size - 1 > 1) {
+            binding.pager.offscreenPageLimit = courseTabItems.size - 1
+        }
+    }
+
+    private fun setupToolbar(hasAccess: Boolean) {
+        binding.toolbar.apply {
+            root.setVisibility(true)
+            collapsedToolbarTitle.text = courseData.course.name
+            courseOrganization.text = courseData.course.org
+            courseTitle.text = courseData.course.name
+            tabs.setVisibility(hasAccess)
+
+            val expiryDate = CourseCardUtils.getFormattedDate(requireContext(), courseData)
+            if (!expiryDate.isNullOrEmpty()) {
+                courseExpiryDate.setVisibility(true)
```

Comment:
IMO, we should follow the practice of using the `apply` method in the kotlin file when setting more than 1 property to an attribute.